Tests/ClientRuntimeTests/ClientRuntimeTests/NetworkingTests/Http/MiddlewareTests/AuthSchemeMiddlewareTests.swift

```diff
@@ -0,0 +1,136 @@
+//
+// Copyright Amazon.com Inc. or its affiliates.
+// All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+import XCTest
+import SmithyTestUtil
+@testable import ClientRuntime
+
+class AuthSchemeMiddlewareTests: XCTestCase {
+    private var contextBuilder: HttpContextBuilder!
+    private var operationStack: OperationStack<MockInput, MockOutput>!
+
+    override func setUp() async throws {
+        try await super.setUp()
+        contextBuilder = HttpContextBuilder()
+            .withAuthSchemeResolver(value: DefaultMockAuthSchemeResolver())
+            .withAuthScheme(value: MockNoAuth())
+            .withIdentityResolver(value: MockIdentityResolver(), type: .aws)
+        operationStack = OperationStack<MockInput, MockOutput>(id: "auth scheme middleware test stack")
+    }
+
+    // Test exception cases
+    func testNoAuthSchemeResolverConfigured() async throws {
+        contextBuilder.attributes.remove(key: AttributeKeys.authSchemeResolver)
+        contextBuilder.withOperation(value: "fillerOp")
+        do {
+            try await AssertSelectedAuthSchemeMatches(builtContext: contextBuilder.build(), expectedAuthScheme: "")
+        } catch ClientError.authError(let message) {
+            XCTAssertEqual(message, "No auth scheme resolver has been configured on the service.")
+        } catch {
+            XCTFail("Expected exception was not thrown.")
```

Comment:
Maybe log the actual error / message here to let the developer know what was actually thrown.

Reply:
(Same in other tests where this pattern is used)

Reply:
Changed as suggested for several tests that this suggestion applies to.